report/5.Conclusion/FutureWork.tex

```diff
@@ -28,3 +28,16 @@ \subsection{Improvements to the language}
 This would make it simpler to implement complex algorithms in HCL, for example sorting algorithms and numerical counting algorithms.
 It would also simplify the process of optimizing naive implementions, especially optimization by dynamic programming.
 
+\textbf{Hash collisions}\\
+As mentioned in section \ref{sec:gencplusplus}, \textbf{generated C++ code}, HCL can potentially generate hash code collisions, when naming types and variables in \textbf{C++}.
+
+There are a couple of potential solutions to this problem.
+One solution would be to only do hashing on individual illegal characters, thus retaining some of the names defined in HCL.
+
+Another solution could be to implement a list of valid names for illegal characters.
+This would mean that \textbf{+} would become \textbf{plus}, \textbf{-} would become minus, and so forth.
+
+\textbf{Generating only the used built-ins}\\
```

Comment:
Well, the c++ compiler will optimize this part. So we could mention that we rely om that to optimize this part